File: scala/digitalwallet/app/src/main/scala/ports/WalletsDatabase.scala
package ports

import core.domain.entities.Wallet
import core.domain.enums.WalletType.WalletType
import core.domain.model.CreateWalletRequest

case class WalletFilter(
 customerId: Option[String] = None,
 walletTYpe:  Option[WalletType] = None,
)

trait WalletsDatabase {
  def insert(request: CreateWalletRequest): Wallet

  def findById(id: String): Wallet

  def find(filter: WalletFilter): List[Wallet]
}

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/ports/WalletsDatabase.kt
package digitalwallet.ports

import digitalwallet.core.domain.enums.WalletType
import digitalwallet.core.domain.models.CreateWalletRequest
import digitalwallet.core.domain.models.Wallet as WalletModel

data class WalletFilter(
    val customerId: String? = null,
    val type: WalletType? = null,
)

interface WalletsDatabase {
    fun insert(request: CreateWalletRequest): WalletModel

    fun findById(id: String): WalletModel?

    fun find(filter: WalletFilter): List<WalletModel>
}
